cmds/core/dhclient/dhclient.go

```diff
@@ -47,51 +48,63 @@ var (
 	v4Port = flag.Int("v4-port", dhcpv4.ServerPort, "DHCPv4 server port to send to")
 )
 
-func main() {
-	flag.Parse()
-	if len(flag.Args()) > 1 {
-		log.Fatalf("only one re")
+func run(args []string, timeout int, retry int, dryRun bool, verbose bool, vverbose bool, ipv4 bool, ipv6 bool, v6Port int, v6Server string, v4Port int) error {
+	if len(args) > 1 {
+		return fmt.Errorf("more than one interface specified")
 	}
 
-	if len(flag.Args()) > 0 {
-		ifName = flag.Args()[0]
+	if len(args) > 0 {
+		ifName = args[0]
 	}
 
 	filteredIfs, err := dhclient.Interfaces(ifName)
 	if err != nil {
-		log.Fatal(err)
+		return err
+	}
+
+	if err := configureAll(filteredIfs, timeout, retry, dryRun, verbose, vverbose, ipv4, ipv6, v6Port, v6Server, v4Port); err != nil {
```

Comment:
Here we are passing 11 arguments from `run()` to `configureAll()` and this functions real purpose is to get the list of filtered interfaces, despite being named `run`.        I wonder if we should consider restructuring this such that `configureAll` is renamed to `run` and then this function gets renamed to something like `filteredInterfaces` or something?

Reply:
I get the reasoning to make function names more specific to the process, but I would prefer to stick to the convention from CONTRIBUTING.md to name the function run() like most other commands:
```
The guideline for testing is to factor out everything "difficult" into a
two-line `main` function which remain untested. For example:

func run(args []string, stdin io.Reader, stdout io.Writer) error {
	...
}

func main() {
	if err := run(os.Args[1:], os.Stdin, os.Stdout); err != nil {
		log.Fatalf("Error: %v", err)
	}
}
```

Reply:
"but I would prefer to stick to the convention from CONTRIBUTING.md to name the function run() like most other commands" - I totally agree! Maybe my intent wasn't clear.

I don't think that the code in `run` today provides much value since the real work is done in `configureAll`.  My proposal is to rename `configureAll` to `run` and move the code in the current `run` to something like `filteredInterfaces`. Example:

```
func run(args []string, opts Opts) error {
  interfaces, err := interfaces(args)
  if err != nil {
    return fmt.Errorf("failed to get interfaces: %v", err)
  }
  // All existing code from configureAll here
}

func interfaces (args []string) ([]netlink.Link, error) {
  // Existing logic from run()
}
```

**Update**: After writing this comment and looking at how the code is structured, I think the existing function signature of `configureAll(ifs []netlink.Link, opts Opts) error` in your code does make sense... so I think I will withdraw this comment :) I don't feel strongly about it.